beanie/odm/utils/pydantic.py

```diff
@@ -1,22 +1,12 @@
 from typing import Any, Type
 
 import pydantic
-from packaging.specifiers import SpecifierSet
-from packaging.version import Version
 from pydantic import BaseModel
 
-
-def is_version_valid(version, requirement):
-    # Parse the requirement as a SpecifierSet
-    specifiers = SpecifierSet(requirement)
-    # Parse the version as a Version
-    v = Version(version)
-    # Check if the version satisfies the specifiers
-    return v in specifiers
-
-
-IS_PYDANTIC_V2 = is_version_valid(pydantic.VERSION, ">=2")
-IS_PYDANTIC_V2_10 = is_version_valid(pydantic.VERSION, ">=2.10")
+IS_PYDANTIC_V2 = int(pydantic.VERSION.split(".")[0]) >= 2
+IS_PYDANTIC_V2_10 = (
+    IS_PYDANTIC_V2 and int(pydantic.VERSION.split(".")[1]) >= 10
+) or int(pydantic.VERSION.split(".")[0]) > 2
```

Comment:
I would remove this "or" part. Any other major Pydantic version would not necessarily be compatible with Beanie and I think we should not bet on this.
Lets make this check do what it says in the name, to explicitly check for Pydantic v2.10.x (or above, constrained to v2).